File: Hard/2026-02-27-3666-Minimum-Operations-to-Equalize-Binary-String/solution.js
var minOperations = function(s, k) {
    const countZeros = (s.match(/0/g) || []).length;
    if (countZeros === 0) return 0;
    if (countZeros < k) return -1;

    let operations = 0;
    while (countZeros > 0) {
        operations++;
        countZeros -= k;
        if (countZeros < 0) return -1;
    }
    return operations;
};

File: Hard/2026-02-27-3666-Minimum-Operations-to-Equalize-Binary-String/solution.py
class Solution:
    def minOperations(self, s: str, k: int) -> int:
        count_zeros = s.count('0')
        if count_zeros == 0:
            return 0
        if count_zeros < k and count_zeros % k != 0:
            return -1
        
        operations = 0
        while count_zeros > 0:
            operations += 1
            count_zeros -= k
            if count_zeros < 0:
                return -1
        
        return operations